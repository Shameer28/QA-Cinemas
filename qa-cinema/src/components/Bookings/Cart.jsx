const Cart = ({getCart, setCart}) => {

    // <button 
    //             onClick={x=>{let newCart = getCart;
    //                     newCart.splice(key, 1);
    //                     setCart(newCart);}}>
    //         X</button>

    if (getCart.length===0){
        return(<></>);
    }else{
        return ( 
            <div>
                <h1>Cart:</h1>
                <p>Name | Movie | Adults, Children | Price </p>
                {getCart.map((x, key)=>(<p key={key}>{x.Name} | {x.MovieTitle} | {x.Adults}, {x.Child} | Â£{x.price}</p>))}
            </div>);
    }

}
 
export default Cart;