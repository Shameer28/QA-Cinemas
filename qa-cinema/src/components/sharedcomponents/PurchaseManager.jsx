import { useState } from "react";
import Payment from "./Payment";
import TicketSelector from "./TicketSelector";

const PurchaseManager = () => {

    const [getCart, setCart] = useState({});

    return ( 
    <div className="text-center">
        <TicketSelector getCart={getCart} setCart={setCart}/>

        <br/> {/* Add Cart? Prop that looks pretty with itemz? */}

        <Payment cart={getCart}/> {/*Maybe a Button to new page*/}
    </div> 
    );
}
 
export default PurchaseManager;