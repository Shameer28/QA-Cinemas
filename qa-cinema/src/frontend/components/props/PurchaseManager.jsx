import { useState } from "react";
import Payment from "./Payment";
import TicketSelector from "./TicketSelector";

const PurchaseManager = () => {

    const [getCart, setCart] = useState({id:0});

    return ( 
    <div>
        <TicketSelector getCart={getCart} setCart={setCart}/>

        <br/> {/* Add Cart? Prop that looks pretty with itemz? */}

        <Payment cart={getCart}/> {/*Maybe a Button to new page*/}
    </div> 
    );
}
 
export default PurchaseManager;