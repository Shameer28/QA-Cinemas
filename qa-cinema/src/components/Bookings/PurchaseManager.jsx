import { useState } from "react";
import TicketSelector from "./TicketSelector";
import StripeCheckoutForm from './StripePayment';
import Cart from "./Cart";
import { Container } from "react-bootstrap";
import { Elements } from "@stripe/react-stripe-js";
import { loadStripe } from '@stripe/stripe-js';

import Carousel from 'react-bootstrap/Carousel'

import image1 from './img/theatre.jpg'
import image2 from './img/set.png'
import image3 from './img/cinemas.jpg'
import image4 from './img/cinema2.jpg'

const stripePromise = loadStripe('pk_test_51JAb8eJE1YNxquuDvcbs6vxzAMjXQfYDU5vS6FvOd35njC6Ceq9VIwWAPn9rK4Z7dMhaGrTyoA41Gw191ltILhIs00yuoTEWIx');


const PurchaseManager = () => {

    const [getCart, setCart] = useState([]);
    const [getPage, setPage] = useState("TicketSelector")

    return (
        <div className="text-center">
            {getPage === "TicketSelector" ? (
                <TicketSelector setPage={setPage} getCart={getCart} setCart={setCart} />
            ) : (
                getPage === "PaymentSuccess" ? (
                    <div className="background">

                        <Carousel controls={false} slide={true} fade={true} pause={false} height="650px">
                            <Carousel.Item interval={3000}>
                                <img
                                    className="d-block w-100"
                                    src={image1}
                                    alt="First slide"
                                />
                            </Carousel.Item>
                            <Carousel.Item interval={3000}>
                                <img
                                    className="d-block w-100"
                                    src={image2}
                                    alt="Second slide"
                                />
                            </Carousel.Item>
                            <Carousel.Item interval={3000}>
                                <img
                                    className="d-block w-100"
                                    src={image3}
                                    alt="Third slide"
                                />
                            </Carousel.Item>
                            <Carousel.Item interval={3000}>
                                <img
                                    className="d-block w-100"
                                    src={image4}
                                    alt="Third slide"
                                />
                            </Carousel.Item>

                        </Carousel>
                        <br></br>
                        <Container >
                            <div style={{ backgroundColor: "#212529", color: "white", border: "solid", borderColor: "black" }}>

                                <h1>Payment Success!</h1>

                            </div>
                        </Container>
                    </div>
                ) : (
                    <Elements stripe={stripePromise}>
                        <StripeCheckoutForm setPage={setPage} getCart={getCart} setCart={setCart} />
                    </Elements>)
            )}

            <div className="background">
                <br></br>
                    <Cart getCart={getCart} setCart={setCart} />
                <br></br>
            </div>
        </div>
    );
}

export default PurchaseManager;