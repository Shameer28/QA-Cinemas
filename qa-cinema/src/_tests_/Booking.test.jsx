import { create } from 'react-test-renderer';
import { fireEvent, render, screen, wait, waitFor} from "@testing-library/react";
import PurchaseManager from '../components/Bookings/PurchaseManager';
import userEvent from "@testing-library/user-event";
import {act} from 'react-dom/test-utils';
import Cart from '../components/Bookings/Cart';

describe(`All Gallery Page testing`, () => {

    it('Should match the snapshot', () => {
        const instance = create(<PurchaseManager/>).toJSON();
        expect(instance).toMatchSnapshot();
    })

    // it('Error Form Alert',async() => {
    //     render(<PurchaseManager/>)
    //     await waitFor(() => screen.getAllByRole("button"))
    //     userEvent.click(screen.getByText("Add to Cart"));
    // })

    it('Fill In Form',  async() =>{
        const instance = render(<PurchaseManager/>);

       

        expect(screen.findByText("Loading Form")).toBeTruthy();

        await waitFor(() => screen.getAllByRole("button")) // Wait for Checkout Buttons

        const textInput = screen.getByLabelText('Name:');
        const textInput1 = screen.getByLabelText('Adults £10:');
        const textInput2 = screen.getByLabelText('Child (Under 16s) £5:');
        const dropInput = screen.getByLabelText('Concession:');

        userEvent.type(textInput, "data");
        userEvent.type(textInput1, "4");
        userEvent.type(textInput2, "3");
        userEvent.selectOptions(dropInput, "2")

        expect(textInput.value).toEqual("data");
        expect(textInput1.value).toEqual("4");
        expect(textInput2.value).toEqual("3");
        expect(dropInput.value).toEqual("2");
        const dropInput1 = screen.getByDisplayValue("Select A Movie")
        expect(dropInput1.value).toEqual("Select A Movie");
        userEvent.selectOptions(dropInput1, "60ec614b2506fc7c20193ce6")
        expect(dropInput1.value).toEqual("60ec614b2506fc7c20193ce6");

        await waitFor(() => screen.getByDisplayValue("25/07/2021 - 9:15"))

        const dropInput2 = screen.getByDisplayValue("25/07/2021 - 9:15")
        expect(dropInput2.value).toEqual("25/07/2021 - 9:15");

        userEvent.click(screen.getByText("Add to Cart"), {target:screen.getByTestId("form").children} ); // event on handle submit is not propergated properly
        
        // Object Added To cart

        await waitFor(() => textInput1=="") // Wait for Cart process

        expect(screen.getByText("Basket:")).toBeTruthy();

        // Payment / Checkout
        userEvent.click(screen.getByText("Checkout"))

        await waitFor(() => screen.getByLabelText("Billing Name:"))


        // Billing Form Details
        const textInput3 = screen.getByLabelText('Billing Name:');
        const textInput4 = screen.getByLabelText('Email:');
        const textInput5 = screen.getByLabelText('Address:');
        
        userEvent.type(textInput3, "Andrew McCall");
        userEvent.type(textInput4, "Email@TestEmail.com");
        userEvent.type(textInput5, "123456789,\nStreet Rd,\nNewcastle");

        expect(textInput3.value).toEqual("Andrew McCall");
        expect(textInput4.value).toEqual("Email@TestEmail.com");
        expect(textInput5.value).toEqual("123456789,\nStreet Rd,\nNewcastle");


    })

    it("Cart - Empty & Full", () =>{

        // Empty Cart
        expect(create(<Cart getCart={[]}/>).toJSON()).toMatchSnapshot();
       
        const instance = create(<Cart getCart={[{
            Adults: "1",
            Child: "1",
            Concessions: "3",
            MovieId: "60ec63782506fc7c20193d1d",
            MovieTimings: "25/07/2021 - 9:15",
            MovieTitle: "Fast and Furious 9",
            Name: "2",
            price: 22.5
        }]}/>);

        // Full Cart
        expect(instance.toJSON()).toMatchSnapshot();
    })

})