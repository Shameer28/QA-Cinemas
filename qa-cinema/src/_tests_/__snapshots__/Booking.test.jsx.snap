// Jest Snapshot v1, https://goo.gl/fbAQLP
exports[`All Gallery Page testing Cart - Empty & Full 1`] = `null`;

exports[`All Gallery Page testing Cart - Empty & Full 2`] = `
<div>
  <div
    className="container"
  >
    <div
      style={
        Object {
          "backgroundColor": "#212529",
          "border": "solid",
          "borderColor": "black",
          "color": "white",
        }
      }
    >
      <div>
        <h1>
          <u>
            Basket:
          </u>
        </h1>
        <br />
        <p>
          <div
            class="center1"
          >
            <h4>
              Name: 
              2
               
              <br />
              Movie Title: 
              Fast and Furious 9
               
              <br />
              Adults: 
              1
              <br />
              Childs: 
              1
               
              <br />
              Snacks: 
              Large Drink & Popcorn
               
              <br />
              Total Price: Â£
              22.5
            </h4>
          </div>
          <br />
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`All Gallery Page testing Should match the snapshot 1`] = `
<div
  className="text-center"
>
  <div
    className="background"
  >
    <div
      className="carousel slide carousel-fade"
      height="650px"
      onKeyDown={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <ol
        className="carousel-indicators"
      >
        <li
          className="active"
          onClick={[Function]}
        />
        <li
          onClick={[Function]}
        />
        <li
          onClick={[Function]}
        />
        <li
          onClick={[Function]}
        />
      </ol>
      <div
        className="carousel-inner"
      >
        <div
          className="active carousel-item"
          interval={3000}
        >
          <img
            alt="First slide"
            className="d-block w-100"
            src="theatre.jpg"
          />
        </div>
        <div
          className="carousel-item"
          interval={3000}
        >
          <img
            alt="Second slide"
            className="d-block w-100"
            src="set.png"
          />
        </div>
        <div
          className="carousel-item"
          interval={3000}
        >
          <img
            alt="Third slide"
            className="d-block w-100"
            src="cinemas.jpg"
          />
        </div>
        <div
          className="carousel-item"
          interval={3000}
        >
          <img
            alt="Third slide"
            className="d-block w-100"
            src="cinema2.jpg"
          />
        </div>
      </div>
    </div>
    <br />
    <h2
      style={
        Object {
          "color": "white",
          "fontWeight": "400",
          "fontfamily": "isonormregular, sans-serif",
          "letterSpacing": ".15em",
          "lineHeight": "1.1",
          "textTransform": "uppercase",
        }
      }
    >
      Book your tickets 
    </h2>
    <br />
    <div
      className="container"
    >
      <div
        className="text-center card"
        style={
          Object {
            "backgroundColor": "#212529",
            "border": "solid",
            "borderColor": "black",
            "color": "white",
          }
        }
      >
        <div
          className="card-header"
        >
          Ticket Select
        </div>
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            <u>
              Please fill out the form to book tickets
            </u>
          </div>
          <p
            className="card-text"
          >
            <h5>
              Loading Form
            </h5>
          </p>
        </div>
        <div
          className="text-muted card-footer"
        />
      </div>
    </div>
  </div>
  <div
    className="background"
  >
    <br />
    <br />
  </div>
</div>
`;
